encoding/ssz/helpers.go

```diff
@@ -113,6 +114,51 @@ func Pack(serializedItems [][]byte) ([][]byte, error) {
 	return chunks, nil
 }
 
+// PackByChunk a given byte array's final chunk with zeroes if needed.
+func PackByChunk(serializedItems [][]byte) ([][bytesPerChunk]byte, error) {
+	areAllEmpty := true
+	for _, item := range serializedItems {
+		if !bytes.Equal(item, []byte{}) {
+			areAllEmpty = false
+			break
+		}
+	}
+	// If there are no items, we return an empty chunk.
+	if len(serializedItems) == 0 || areAllEmpty {
+		emptyChunk := [bytesPerChunk]byte{}
+		return [][bytesPerChunk]byte{emptyChunk}, nil
+	} else if len(serializedItems[0]) == bytesPerChunk {
+		// If each item has exactly BYTES_PER_CHUNK length, we return the list of serialized items.
+		chunks := make([][bytesPerChunk]byte, 0, len(serializedItems))
+		for _, c := range serializedItems {
+			chunks = append(chunks, bytesutil.ToBytes32(c))
+		}
+		return chunks, nil
+	}
+	// We flatten the list in order to pack its items into byte chunks correctly.
+	var orderedItems []byte
+	for _, item := range serializedItems {
+		orderedItems = append(orderedItems, item...)
+	}
+	numItems := len(orderedItems)
+	var chunks [][bytesPerChunk]byte
+	for i := 0; i < numItems; i += bytesPerChunk {
+		j := i + bytesPerChunk
+		// We create our upper bound index of the chunk, if it is greater than numItems,
+		// we set it as numItems itself.
+		if j > numItems {
+			j = numItems
+		}
```

Comment:
The right padding here is a little subtle and maybe worthy of comment. It works because `ToBytes32` allocates a `[32]byte` (which has a zero value of 32 empty-bytes) and then copies the slice into that value. This condition results in the length of the slice passed to `ToBytes32` to be less than 32 bytes for the final iteration over `orderedItems` when `len(orderedItems) % 32 != 0`.